source/agora/node/Network.d

```diff
@@ -70,7 +70,8 @@ class Network
 
     /// Ctor
     public this (in NodeConfig node_config, in string[] peers)
-    {
+    in { assert(peers.length > 0, "No network option found"); }
+    do {
```

Comment:
`do`? I thought this was supposed to be `body`?

Reply:
ah they changed it, didn't they..

Reply:
There was some DIP that simplified `in` expressions, but I'm not sure if it was approved or implemented. It would avoid these ugly braces. :)

Reply:
It's implemented (but its ugly and not all editors support it)

Reply:
What is the actual syntax?

Reply:
https://dlang.org/spec/contracts.html#pre_post_contracts

Reply:
ah so it would be something like:

```D
public this ( in string[] peers) 
    in(peers.length > 0, "No peers")
{
}
```

Similar to template constraints.